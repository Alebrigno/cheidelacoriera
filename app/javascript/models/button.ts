export interface Button {
  className: string;
  url: string;
  text: string;
}
