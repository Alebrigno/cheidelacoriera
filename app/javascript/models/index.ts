export { Button } from "./button";
export { Game } from "./game";
export { Team } from "./team";
export { TransportMean } from "./transportMean";
export { User } from "./user";
