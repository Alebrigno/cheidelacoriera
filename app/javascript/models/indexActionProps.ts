export interface IndexActionProps {
    body: string;
    buttonText: string;
    heading: string;
    iconClass: string;
    name: string;
    url: string; 
  }
  