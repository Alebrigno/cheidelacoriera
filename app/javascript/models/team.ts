export interface Team {
  imageUrl: string;
  name: string;
}
