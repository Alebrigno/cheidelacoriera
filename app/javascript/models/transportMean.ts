export interface TransportMean {
  iconClass?: string;
  name: string;
}