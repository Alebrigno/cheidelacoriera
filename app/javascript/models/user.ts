export interface User {
    canBook?: boolean;
}