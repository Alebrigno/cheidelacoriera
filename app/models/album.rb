# frozen_string_literal: true

class Album < ApplicationRecord
  validates :title, presence: true
  validates :url, http_url: true, allow_nil: true

  belongs_to :event, inverse_of: :albums, optional: true

  def to_s
    title
  end
end
