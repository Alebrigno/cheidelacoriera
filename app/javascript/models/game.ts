import { Button } from "./button";
import { Team } from "./team";
import { TransportMean } from "./transportMean";
import { User } from "./user";

interface GameTeams {
  away: Team;
  home: Team;
}

export interface Game {
  availability: number;
  audience?: string;
  buttons?: Button[];
  competition: string;
  currentUser?: User;
  date: string;
  id: number;
  score?: string;
  status?: string;
  teams: GameTeams;
  time: string;
  transportMean: TransportMean;
}