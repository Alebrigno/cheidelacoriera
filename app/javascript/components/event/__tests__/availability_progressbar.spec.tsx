import { configure } from "enzyme";
import * as Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });
import { EventAvailabilityProgressbar } from "..";
import { Game } from "../../../models";

describe("EventAvailabilityProgressbar", () => {
  [
    { availability: 100 },
    { availability: 100, currentUser: { canBook: false } },
    { availability: 1, currentUser: { canBook: true } },
    { availability: 26, currentUser: { canBook: true } },
    { availability: 51, currentUser: { canBook: true } },
    { availability: 76, currentUser: { canBook: true } },
  ].forEach((game: Partial<Game>) => {
    test(`when availability: ${game.availability}, currentUser: ${
      game.currentUser && JSON.stringify(game.currentUser)
    }`, () => {
      const progressBar = new EventAvailabilityProgressbar({
        game: game,
      });
      expect(progressBar.render()).toMatchSnapshot();
    });
  });
});
