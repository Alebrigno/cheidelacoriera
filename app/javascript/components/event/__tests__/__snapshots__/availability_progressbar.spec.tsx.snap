// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventAvailabilityProgressbar when availability: 1, currentUser: {"canBook":true} 1`] = `
<div
  className="progress event-progress"
>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-danger"
    role="progressbar"
    style={
      Object {
        "width": "99%",
      }
    }
  >
    <span
      className="sr-only"
    >
      99%
    </span>
  </div>
</div>
`;

exports[`EventAvailabilityProgressbar when availability: 26, currentUser: {"canBook":true} 1`] = `
<div
  className="progress event-progress"
>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-warning"
    role="progressbar"
    style={
      Object {
        "width": "74%",
      }
    }
  >
    <span
      className="sr-only"
    >
      74%
    </span>
  </div>
</div>
`;

exports[`EventAvailabilityProgressbar when availability: 51, currentUser: {"canBook":true} 1`] = `
<div
  className="progress event-progress"
>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-info"
    role="progressbar"
    style={
      Object {
        "width": "49%",
      }
    }
  >
    <span
      className="sr-only"
    >
      49%
    </span>
  </div>
</div>
`;

exports[`EventAvailabilityProgressbar when availability: 76, currentUser: {"canBook":true} 1`] = `
<div
  className="progress event-progress"
>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar progress-bar-success"
    role="progressbar"
    style={
      Object {
        "width": "24%",
      }
    }
  >
    <span
      className="sr-only"
    >
      24%
    </span>
  </div>
</div>
`;

exports[`EventAvailabilityProgressbar when availability: 100, currentUser: {"canBook":false} 1`] = `
<div
  className="progress event-progress"
/>
`;

exports[`EventAvailabilityProgressbar when availability: 100, currentUser: undefined 1`] = `
<div
  className="progress event-progress"
/>
`;
